services/chronikService.ts

```diff
@@ -159,11 +179,12 @@ export function toHash160 (address: string): {type: ScriptType, hash160: string}
   }
 }
 
-export function outputScriptToAddress (outputScript: String): string | boolean {
-  // returns P2SH or P2PKH address
+export function outputScriptToAddress (outputScript: String | undefined): string {
+  // returns P2SH or P2PKH address, empty string otherwise
   // P2PKH addresses are in outputScript of type 76a914...88ac
   // P2SH addresses are in outputScript of type a914...87
-  // Return false if cannot determine P2PKH or P2SH address
+
+  if (outputScript == null) return ''
```

Comment:
some things here:
- If it can be `undefined` why check for `null`?
- what's the reasoning for expecting undefined values and returning the empty string when they exist?

Reply:
In better words: why do we expect inputs or outputs to have no address associated with them? Are these 'mine transactions'? If that's the case, seems better to me to let `address` be optional and explicitly setting it as null instead of as an empty string. If that's only possible for outputs and not for inputs (or vice versa), then I think it would be better to have both interfaces instead of a unified one.

Reply:
1. changed verify  `undefined` instead of `null`
2. changed to return `undefined` instead of `''`, indeed the inputs can have no address, I assume being the coins transferred from Coinbase (mined), however I chose to allow undefined in the output's address as well because the `outputScriptToAddress` function can return undefined if address is not a P2PKH or P2SH format. I imagine this could happen, so better be prepared.